lib/puppet/util/log/destinations.rb

```diff
@@ -164,6 +164,7 @@ def handle(msg)
     }
 
     str = msg.respond_to?(:multiline) ? msg.multiline : msg.to_s
+    str = str.encode(Encoding.default_external)
```

Comment:
This is caused by debug output generated at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/user/windows_adsi.rb#L61 - it's trying to smash together strings of different encodings.

It's possible that Ruby's COM support is handing back strings that have been re-encoded from `UTF-16LE` to the local codepage, and we're later trying to smash them together unsuccessfully with `UTF-8`.

Note that any time our internal COM helpers read memory for strings, we treat it as `UTF-16LE` (i.e. for task scheduler - https://github.com/puppetlabs/puppet/blob/6b318af97f4254fd4b4f04583b9b677f8f396176/lib/puppet/util/windows/api_types.rb#L56-L81), but Ruby has a bad habit of wanting to convert API results into local codepage all the time.